### PR TITLE
Add development environment docs

### PR DESCRIPTION
**Pull request by BryceGattis**
This PR adds a new section of the CONTRIBUTING.md file that helps explain how to set up your development environment so that you can contribute to Rez more easily.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+
+### Setting Up
+
+To begin development on Rez you'll first need to set up your development environment. There are many 
+ways you can do it, but this is the recommended approach.
+
+1. Fork the repo and clone it.
+2. Create a new Git branch and check it out.
+3. Create a virtual environment in the same directory as the repo by running `python -m venv venv`.
+4. Pip install your local Rez code by running `pip install .`
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:18]: The virtual environment needs to be activated. It should also be noted of what's the different between this and installing using the `install.py` script.

In fact, I would encourage users to use the install script first, and then only mention `pip install .` later. The reason being that not all tests will be run if you `pip install .`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,5 +1,29 @@
 # Contributing To Rez
 
+## Development Environment
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:3]: I'd move this under the "guidelines" section I think.

**Review by JeanChristopheMorinPerso — COMMENTED** [review]
Thanks @BryceGattis, I left a couple of notes. Let me know if you have questions.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -21,6 +23,38 @@ sending a pull request. Please follow these guidelines:
...
+1. Fork the repo and clone it.
+2. Create a new Git branch and check it out.
+3. Create a virtual environment in the same directory as the repo by running `python -m venv venv`.
+4. Activate the virtual environment by running the `activate` file.
+5. Pip install your local Rez code by running `pip install .`
+
+### Running Tests
+
+1. Set up your development environment as shown above.
+2. Run `rez selftest`.
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:56]: When installing with the install script, this won't work. The rez command won't be in `$PATH`.

Comment (BryceGattis) [reply CONTRIBUTING.md:56]: @JeanChristopheMorinPerso Yeah I see when you pip install Rez the binaries get placed directly in the "Scripts" folder on Windows, whereas when using Rez's `install.py` file, Rez binaries are placed in the "Scripts/rez" folder, so they are not automatically found by just activating the virtual environment :/

I guess I can just add another step to the `install.py` method that mentions you need to manually add the "Scripts/rez" folder to $PATH?

Comment (JeanChristopheMorinPerso) [reply CONTRIBUTING.md:56]: Yep! I would also mention that we recommend running the tests with `pytest` and the `parameterized` package.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -21,6 +23,40 @@ sending a pull request. Please follow these guidelines:
...
+environment for you, patch the Rez binaries, and copy completion scripts. All tests will be run this way.
+
+1. Fork the repo and clone it.
+2. Create a new Git branch and check it out.
+3. Install your local Rez code by running `python install.py venv`.
+4. Activate the virtual environment by running the `activate` file.
+5. Add the `Scripts/rez` folder on Windows or the `bin/rez` folder on Mac/Linux to the `PATH` environment variable.
+
+There is an alternative method of setting up your development environment, that doesn't use the `install.py`
+script. This method is not recommended however, as not all tests will be run this way.
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:46]: I wouldn't say that it's not recommended. It's more that it won't run all tests. So it's useful to test quick things, etc. We can also easily use an editable install `pip isntall -e .` which speeds up the development a little bit.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -21,6 +23,40 @@ sending a pull request. Please follow these guidelines:
...
+1. Fork the repo and clone it.
+2. Create a new Git branch and check it out.
+3. Create a virtual environment in the same directory as the repo by running `python -m venv venv`.
+4. Activate the virtual environment by running the `activate` file.
+5. Pip install your local Rez code by running `pip install .`
+
+### Running Tests
+
+1. Set up your development environment as shown above.
+2. Install the `pytest` and `parameterized` Python packages to your virtual environment for improved test output.
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:57]: It wouold be worth mentionning how to do this (`pip install pytest parameterized`)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -21,6 +23,45 @@ sending a pull request. Please follow these guidelines:
...
+## Development Environment
+
+### Prerequisites 
+
+On Windows, make sure to set your PowerShell execution policy as shown [here](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-5.1),
+otherwise the PowerShell tests will fail.
+
+### Setting Up
+
+To begin development on Rez you'll first need to set up your development environment. There are many
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:35]:
```suggestion
To begin development on rez you'll first need to set up your development environment. There are many 
```
rez needs a lower `r` :)

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -21,6 +23,45 @@ sending a pull request. Please follow these guidelines:
...
+
+To begin development on Rez you'll first need to set up your development environment. There are many 
+ways you can do it, but these are the recommended approaches.
+
+This first approach will automatically create a virtual environment for you, patch the Rez binaries, 
+and copy completion scripts. All tests will be run this way.
+
+1. Fork the repo and clone it.
+2. Create a new Git branch and check it out.
+3. Install your local Rez code by running `python install.py venv`.
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:43]:
```suggestion
3. Install your local rez code by running `python install.py venv`.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -21,6 +23,45 @@ sending a pull request. Please follow these guidelines:
...
+4. Activate the virtual environment by running the `activate` file.
+5. Add the `Scripts/rez` folder on Windows or the `bin/rez` folder on Mac/Linux to the `PATH` environment variable.
+
+There is an alternative method of setting up your development environment, that doesn't use the `install.py`
+script. Please note that not all tests will be run if Rez is installed this way.
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:48]:
```suggestion
script. Please note that not all tests will be run if rez is installed this way.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -21,6 +23,45 @@ sending a pull request. Please follow these guidelines:
...
+
+1. Fork the repo and clone it.
+2. Create a new Git branch and check it out.
+3. Create a virtual environment in the same directory as the repo by running `python -m venv venv`.
+4. Activate the virtual environment by running the `activate` file.
+5. Pip install your local Rez code by running `pip install .`.
```
Comment (JeanChristopheMorinPerso) [inline CONTRIBUTING.md:54]:
```suggestion
5. Pip install your local rez code by running `pip install .`.
```

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Thank you @BryceGattis!

**Pull request merged**